doc/branch.rst

```diff
@@ -85,47 +81,139 @@ as a branch if one of its choices is excluded::
 Because the ``else`` clause is excluded, the ``if`` only has one possible next
 line, so it isn't considered a branch at all.
 
-
 Structurally partial branches
 -----------------------------
 
 Sometimes branching constructs are used in unusual ways that don't actually
-branch.  For example::
+branch. For example::
 
     while True:
         if cond:
             break
         do_something()
 
-Here the while loop will never exit normally, so it doesn't take both of its
-"possible" branches.  For some of these constructs, such as "while True:" and
-"if 0:", coverage.py understands what is going on.  In these cases, the line
-will not be marked as a partial branch.
+Here the ``while`` loop will never exit normally, so it doesn't take both of
+its "possible" branches. For some of these constructs, such as ``while True:``
+and ``if 0:``, coverage.py understands what is going on.
 
-But there are many ways in your own code to write intentionally partial
-branches, and you don't want coverage.py pestering you about them.  You can
-tell coverage.py that you don't want them flagged by marking them with a
-pragma::
+But there are many ways to intentionally write partial branches. You can mark
+them with a pragma::
 
     i = 0
     while i < 999999999:  # pragma: no branch
         if eventually():
             break
 
-Here the while loop will never complete because the break will always be taken
-at some point.  Coverage.py can't work that out on its own, but the "no branch"
-pragma indicates that the branch is known to be partial, and the line is not
-flagged.
-
 Generator expressions
-=====================
```

Comment:
This was meant to be a section within "Structurally partial branches", which is why it has a different header rule style.